pkg/cloud/cloud_test.go

```diff
@@ -110,7 +109,7 @@ func TestNewCloud(t *testing.T) {
 		},
 	}
 	for _, tc := range testCases {
-		ec2Cloud, err := NewCloud(tc.region, tc.awsSdkDebugLog, tc.userAgentExtra, tc.batchingEnabled)
+		ec2Cloud, err := NewCloud(tc.region, tc.awsSdkDebugLog, tc.userAgentExtra, tc.batchingEnabled, "")
```

Comment:
can we also add test case where roleArn is not empty

Reply:
I'm not sure what it will test? It just is a struct that is attached to `aws.Config.Credentials` which are used when API calls are made. This test is just tests are just testing conversions from internal package representation of `cloud` object to `Cloud` interface which is exposed for other packages in the directory.

Reply:
Yeah, I will need some feedback from connor on this because all of this is just setting the providers, the actual calls are not being made for me to mock.